src/libs/intents/localMethods.ts

```diff
@@ -91,27 +91,43 @@ export const internalMethods = {
     setFlagshipUI(
       intent,
       EnvService.nameIs(strings.environments.test)
-        ? internalMethods.setFlagshipUI.caller?.name // eslint-disable-line @typescript-eslint/no-unnecessary-condition
+        ? // @TODO: find why the function caller was determined to be possibly undefined
+          // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+          internalMethods.setFlagshipUI.caller?.name
         : ''
     )
 }
 
+interface CustomMethods {
+  fetchSessionCode: () => Promise<string | null>
+  showInAppBrowser: (args: { url: string }) => Promise<BrowserResult>
+}
+
+/**
+ * For now cozy-intent doesn't accept methods resolving to void.
+ * We can use this wrapper to still execute an async method an resolve to null no matter what.
+ */
+const nativeMethodWrapper =
+  <T extends () => Promise<void>>(method: T) =>
+  async (): Promise<null> => {
+    await method()
+
+    return null
+  }
+
 export const localMethods = (
   client: CozyClient | undefined
-): NativeMethodsRegister => {
+): NativeMethodsRegister | CustomMethods => {
```

Comment:
Why is this new type needed?

Reply:
It should be added into cozy-intent NativeMethodsRegister when I have some time. I think it's ok to have this temporarily though